### PR TITLE
Adding support for relative url location

### PR DESCRIPTION
**Pull request by swiftugandan**
* fixes conflicts from #80
* Also prevents caching of dynamic responses

I would really love to get this reviewed and possibly merged into master to avoid having to fix conflicts with master each time a new change is merged.

### CODE REVIEW

File: src/server/kiwix-serve.cpp [hunk 1]
```diff
@@ -423,14 +425,14 @@ static struct MHD_Response* handle_suggest(RequestContext* request_context)
   mimeType = "application/json; charset=utf-8";
   bool deflated = request_context->acceptEncodingDeflate && compress_content(content, mimeType);
   return build_response(
-      content.data(), content.size(), "", mimeType, deflated, true);
+      content.data(), content.size(), "", mimeType, deflated, false);
```
Comment (mgautierfr) [inline src/server/kiwix-serve.cpp]: Why this ?

The content of the response is entirely defined by the request (the name of the zim and the `term` to search for), so we can cache it.

The only way the content change with the same url is that someone change the zim but keep the same name as before. And if we want to handle this case, we should not cache any responses.

File: src/server/kiwix-serve.cpp [hunk 2]
```diff
@@ -505,7 +507,7 @@ static struct MHD_Response* handle_search(RequestContext* request_context)
                         httpRedirection,
                         mimeType,
                         deflated,
-                        true);
+                        false);
```
Comment (mgautierfr) [inline src/server/kiwix-serve.cpp]: Same here.

File: src/server/kiwix-serve.cpp [hunk 3]
```diff
@@ -624,7 +626,7 @@ static struct MHD_Response* handle_default(RequestContext* request_context)
 
   bool deflated = request_context->acceptEncodingDeflate && compress_content(content, mimeType);
   return build_response(
-      content.data(), content.size(), "", mimeType, deflated, true);
+      content.data(), content.size(), "", mimeType, deflated, false);
```
Comment (mgautierfr) [inline src/server/kiwix-serve.cpp]: And here :)

File: src/server/kiwix-serve.cpp (hunk 1 above)
Comment (swiftugandan) [reply src/server/kiwix-serve.cpp]: makes sense, let me fix this, thank you

File: src/server/kiwix-serve.cpp (hunk 3 above)
Comment (swiftugandan) [reply src/server/kiwix-serve.cpp]: I think the default page should not be cached, b'se when one adds a new zim to the library, we want the welcome page to show the updates list, and not the cached site.

Comment (mgautierfr) [reply src/server/kiwix-serve.cpp]: You're totally right here. Good catch.

**Comment by mgautierfr** [comment]
Your last commit said that it disable caching for the welcome page but what it does is re-enabling caching for other pages (revert some change from the first commit).

Can you change your PR to have a first commit changing things related to the root_location and a second commit disabling the caching of the welcome page ?
If not, I can do it without problem (I even already done it) but I prefer give you the chance to change it yourself, it is your PR after all :)

**Comment by swiftugandan** [comment]
@mgautierfr, please help make the PR changes. Thanx

**Comment by mgautierfr** [comment]
I've just create a new PR #86 with the rewrite of you commit.

No change at all on the code change itself. It is just a "resplit" of the commits.
As I have no write access on @swiftugandan's repository, I had to create a new PR.

So, I'm closing this one and will merge the new one.

Thanks a lot @swiftugandan for this PR.

**Pull request closed without merging**